### ISSUE TITLE
Threat model: Objectively discuss risks & benefits of decentralized approach

### ISSUE THREAD

**Issue by adewes** [DP-3T/documents#166]
Reading the security and main whitepaper it seems that you clearly favor a decentralized approach to contact tracing, which is of course a reasonable choice. I think it would be fair though to clarify the different trade-offs and risks of a centralized vs. a decentralized approach more clearly. For example, I see the following main difference in risk:

- In a fully decentralized approach, the matching of infection IDs takes part on the client. This means that clients need to be able to obtain a full list of IDs of infected people, which opens up several attack scenarios that have been discussed in the whitepaper and some other Github issues already.
- In a partly centralized approach, clients would upload their seen IDs to a central authority, which would then check them against infected IDs and report a simple yes/no answer. This model prevents some attack scenarios from the decentralized approach but of course now puts the central authority in a position to abuse this data.

Your approach seems to relatively allocate more trust to the clients as compared to the central authority. While I think that might be a reasonable stance I also think it would be good to mention and explain the trust allocation more explicitly, as it would otherwise just be a matter of "taste" whether to choose a centralized or decentralized approach.

**Comment by lbarman** [comment]
Hi @adewes, thanks for your input! 
We agree, both approaches have pros and cons. You saw pages 29-32 of the white paper, which is precisely this comparison; I assume your issue is notably with section 6.3 which is indeed very short and not acknowledging a range of attacks on the decentralized system (all of which should be acknowledged pages 22-28, by the way - but they should also be in the comparison). Anything else that I'm missing ?

**Comment by kugelfish42** [comment]
I guess the biggest material difference is that with a centralized approach the health authorities learn a window of the contact graph of every infected person who shares this data while in a distributed model any motivated/resourceful attacker can de-anonymize the identity of some number of infected persons.

The health authorities getting to know the contact graph can be both a pro for effectiveness and a con for privacy. Human contact tracers could join this data with other sources (e.g. interviews) and dismiss some contacts as irrelevant (i.e. through the wall of adjacent apartments) leading to less false alarms. On the other hand, this data would clearly also enable police state surveillance in unanticipated ways. In the distributed model there is little room for human judgement to filter out noise and false positives and if the system produces too many irrelevant contacts, users might quickly start to ignore it.

The BlueTrace whitepaper makes some arguments in favor of a centralized approach: https://bluetrace.io/static/bluetrace_whitepaper-938063656596c104632def383eb33b3c.pdf

**Comment by kennypaterson** [comment]
@kugelfish42 This is essentially correct, except in one detail: against a fully centralised system, a very determined and well-resourced attacker can also identify infected persons by doing an "identity cycling" attack. Here, the idea is to switch frequently between multiple registrations done with different identities/devices; when you receive a notification that one of these registrations has been "exposed" then you can look up its history and see when/where this registration was used, and hence infer who the infected party was. It's hard to know for sure whether or not such an attack would work against a concrete centralised system because, well, we haven't seen details of any such system yet... 

Does this make sense?

**Comment by kugelfish42** [comment]
@kennypaterson - There might be an even easier way to get the list of infected persons. Presumably this list already exists (maybe in paper form) somewhere in various health authorities. A well-resourced attacher could get them from there. In addition a centralize system also has a key-store which allows to map observable tags to real-world identities for all participants (not just identified persons). This would allow de-anonymizing anybody who participates in the system.

Presumably at least one centralized system has existed for a while - the one operated by the Government of Singapore.

Another way to look at it would be what are the most costly failure modes in each case? For a centralized system, that might be the risk of police-state abuse who met whom when (which they could also get through traditional means), for a distributed one, that might be the risk of being ineffective.

**Comment by kennypaterson** [comment]
@kugelfish42 Thanks. Indeed, these are the kinds of considerations we had to take into account when settling on our decentralised approach for DP-3T. We had many debates internally. There are hard trade-offs and no system is perfect. In the end, we decided it was essential to guard against facilitating the creation of mass surveillance system. It would be interesting to see a full analysis of the centralised solutions that are being promoted elsewhere. The Singapore app has now been open-sourced, but we have not had time to analyse it in detail. Other centralised initiatives have so far been less open in their design approach, but will eventually have to be so, otherwise they run the risk of not being accepted by civil society.
